src/core/source_buffers/queued_source_buffer.ts

```diff
@@ -380,7 +388,11 @@ export default class QueuedSourceBuffer<T> {
     try {
       switch (queueItem.type) {
         case SourceBufferAction.Append:
-          const { segment, timestampOffset = 0 } = queueItem.args;
+          const { segment, timestampOffset = 0, codec } = queueItem.args;
+          if (this._currentCodec !== codec) {
+            tryToChangeSourceBufferType(this._buffer, codec);
+            this._currentCodec = codec;
+          }
```

Comment:
Where the magic happens

Reply:
We shouldn't change the currentCodec when we couldn't change sourceBuffer codec.

Reply:
fixed?

Reply:
:+1: